stories/core/ExampleComponents/FreeDrawExample.js

```diff
@@ -12,37 +12,49 @@ export default class FreeDrawExample extends PureComponent {
     onDrawBegin: PropTypes.func,
     dataStyle: PropTypes.object,
     theme: PropTypes.object,
+    onDrawEnd: PropTypes.func,
+    allowMultipleShapes: PropTypes.bool,
   }
 
   static defaultProps = {
     theme: {},
     shapes: {},
+    allowMultipleShapes: false,
   }
 
-  constructor(props) {
-    super(props)
-    this.state = {
-      shapes: this.props.shapes,
-      enabled: false,
-    }
+  state = {
+    shapes: this.props.shapes,
+    enabled: this.props.allowMultipleShapes,
   }
 
   @autobind
   handleOnDrawEnd(shape) {
+    const { shapes } = this.state
+
+    if (this.props.onDrawEnd) this.props.onDrawEnd(shape)
+    const shapeId = Object.keys(shapes).length
     this.setState({
-      shapes: shape && { 0: shape },
-      enabled: false,
+      shapes: shape && { ...shapes, [shapeId]: shape },
+      enabled: this.props.allowMultipleShapes,
     })
   }
 
   @autobind
   handleClick() {
-    this.setState({ enabled: !this.state.enabled })
+    const existingShapes = this.props.shapes
+    const { enabled, shapes: _ } = this.state
+
+    const shapes = !enabled && Object.keys(_).length > 0 && _ !== existingShapes
```

Comment:
`Object.keys(_).length > 0` should just be `Object.keys(_).length`

Reply:
I also don't think we should really be using both props and state to write conditionals.  What happens if `props.shapes` changes is that going to be an issue?

Reply:
I think I'm going to take out the state shape logic - I think it was intended to store the shapes and now since we are trying to reconcile both it's getting out of hand. I'll just reset the state on click back to the props passed in.